scripts/staging/fm_regression.dml

```diff
@@ -0,0 +1,91 @@
+#-------------------------------------------------------------
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+# 
+#   http://www.apache.org/licenses/LICENSE-2.0
+# 
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+#-------------------------------------------------------------
+
+/**
+ * Factorization Machines for Regression.
+ */
+
+# Imports
+source("algorithms/fm.dml") as fm
+source("nn/layers/l2_loss.dml") as l2_loss
+
+train = function(matrix[double] X)
+    return (matrix[double] out) {
+   /*
+    * Trains a FM model.
+    */
+    
+    # 1.initialize fm core
+    [w0, W, V] = fm::init(d, k);
+    
+    # 2.initialize adam optimizer
+    [mw0, vw0] = adam::init(w0);
+    [mW, vW] = adam::init(W);
+    [mV, vV] = adam::init(V);
+    
+    # 3.Send inputs through fm::forward
+    y = fm::forward(X, w0, W, V);
+    
+    # 4.Send the above result through log_loss::forward
+    sfy = sigmoid::forward(y);
+    
+    # 5.Send the above result through fm::backward
```

Comment:
before we can use `fm::backward`, we need to compute gradients from a loss (l2_loss in this case) using `l2_loss::backward` similar to the `log_loss::backward` in the classification case.  then send that through `sigmoid::backward`, and then through `fm::backward`.  same idea as the classification, just a different loss function that we are optimizing.